src/Heist/Extra/Splices/Pandoc/Render.hs

```diff
@@ -253,11 +253,44 @@ rpInline' ctx@RenderCtx {..} i = case i of
         w <&> \nodes ->
           [X.TextNode "“"] <> nodes <> [X.TextNode "”"]
 
+-- | Convert raw html attribute sequence into span
+--
+-- For example, this markdown: `<kbd>ctrl</kbd>`, which is in native pandoc:
+--   RawInline (Format "html") "<kbd>" : Str "ctrl" : RawInline (Format "html") "</kbd>"
+-- … is converted as:
+--   <span xmlhtmlraw=""><kbd>ctrl</kbd></span>
+-- … instead of the default behavior which is:
+--   <span xmlhtmlraw=""><kbd></kbd></span>ctrl<span xmlhtmlraw=""></span>
+convertRawInline :: [B.Inline] -> [B.Inline] -> [B.Inline]
+convertRawInline acc = \case
+  [] -> reverse acc
+  (B.RawInline (B.Format "html") oTag : rest)
+    | -- This is a new raw tag, let's find a matching closing tag
+      Just (newElem, is) <- mkHtmlInline oTag rest
+        -> convertRawInline (newElem : acc) is
+  i : is -> convertRawInline (i : acc) is
+ where
+   mkHtmlInline :: Text -> [B.Inline] -> Maybe (B.Inline, [B.Inline])
+   mkHtmlInline oTag rest = case span (not . isClosingTag oTag) rest of
+     -- Collect the inner element until a closing tag
+     (inner, (closing : is))
+       | -- Verify we did find a matching tag
+         isClosingTag oTag closing ->
+           let inner' = oTag <> plainify inner <> "</" <> T.drop 1 oTag
+            in Just (B.RawInline (B.Format "html") inner', is)
+     _ -> Nothing
+
+   isClosingTag :: Text -> B.Inline -> Bool
+   isClosingTag oTag = \case
+     B.RawInline (B.Format "html") eTag ->
+       "<" `T.isPrefixOf` oTag && "</" `T.isPrefixOf` eTag && T.drop 1 oTag == T.drop 2 eTag
+     _ -> False
+
 -- | Like rpInline', but supports task checkbox in the given inlines.
 rpInlineWithTasks :: RenderCtx -> [B.Inline] -> HI.Splice Identity
 rpInlineWithTasks ctx is =
   rpTask ctx is $
-    rpInline ctx `foldMapM` is
+    rpInline ctx `foldMapM` convertRawInline [] is
```

Comment:
I just noticed that this is being called from `rpInlineWithTasks` which kind of violates the purose of this function.

When I read the surrounding code, I realize that this is theonly function that transforms on `[B.Inline]`. Also, it is called only for `B.Plain` and `B.Para`:

![image](https://github.com/user-attachments/assets/e92ed6eb-46ca-42c4-9589-aaf07da056ae)

And not called elsewhere, like `B.BulletList`. Is that an issue, or does Plain and Para cover all cases?

In any case, it looks like we should rename `rpInlineWithTasks` to something suitable - as this function now seems to do some "processing" of groups of inlines, doing two things now: process task lists and handle the raw inline issue.

Reply:
Good catch, I guess we should apply `rpInlineWithTasks` to every Pandoc.Inline.